src/components/Header/Header.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  memo,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { AuthContext } from '../Auth/AuthContext';
+
+interface Props {
+  onAddTodo: (fieldsForCreate: Omit<Todo, 'id'>) => Promise<void>;
+  showError: (message: string) => void;
+  onChangeAllTodos: () => void;
+  isAllTodosCompleted: boolean;
+  isAddingTodo: boolean,
+}
+
+export const Header: React.FC<Props> = memo(({
+  onAddTodo,
+  showError,
+  onChangeAllTodos,
```

Comment:
or `handleChangeAllTodos`
```suggestion
  changeAllTodos,
```